js/search.js

```diff
@@ -0,0 +1,99 @@
+class searchResult {
+  constructor(element) {
+    this.element = element;
+    this.#build();
+  }
+
+  #build() {
+    this.div = document.createElement("div");
+    Object.assign(this.div, {
+      class: "list-group list-group-flush",
+      id: "resultList",
+    });
+    this.div.setAttribute("class", "list-group list-group-flush");
+
+    this.template = document.createElement("template");
+    this.template.setAttribute("id", "template");
+
+    this.img = document.createElement("img");
+    this.img.setAttribute("class", "imageCompany");
+    this.a = document.createElement("a");
+    this.a.setAttribute("class", "list-group-item");
+    this.h = document.createElement("h6");
+    this.aSpan = document.createElement("span");
+    this.aSpan.setAttribute("id", "percentage");
+
+    this.divResult = document.createElement("div");
+    this.divResult.setAttribute("id", "resultList");
+
+    this.a.appendChild(this.img);
+    this.a.appendChild(this.h);
+    this.a.appendChild(this.aSpan);
+    this.template.content.appendChild(this.a);
+    this.div.appendChild(this.template);
+    this.div.appendChild(this.divResult);
+    this.element.appendChild(this.div);
+  }
+  async getApi(url) {
+    try {
+      this.response = await fetch(url);
+      this.result = await this.response.json();
+      return this.result;
+    } catch (err) {
+      console.log("err: ", err);
+    }
+  }
+  async getResult(apiResult) {
+    this.divResult.innerHTML = "";
+
+    apiResult.forEach(async (element) => {
+      this.companyInfo = await this.getCompanyInfo(element.symbol);
```

Comment:
In order to recive al lthe results in the same time, you should use Promies.all() here